src/modules/credential/cert/path_certs.rs

```diff
@@ -0,0 +1,440 @@
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use derive_more::{Deref, DerefMut};
+use openssl::x509::X509;
+use openssl_sys::XKU_SSL_CLIENT;
+use serde::{Deserialize, Serialize};
+
+use super::{CertBackend, CertBackendInner};
+use crate::{
+    context::Context,
+    errors::RvError,
+    logical::{field::FieldTrait, Backend, Field, FieldType, Operation, Path, PathOperation, Request, Response},
+    new_fields, new_fields_internal, new_path, new_path_internal, rv_error_response,
+    storage::StorageEntry,
+    utils::{
+        cert::{
+            deserialize_vec_x509, has_x509_ext_key_usage, has_x509_ext_key_usage_flag, is_ca_cert, serialize_vec_x509,
+        },
+        deserialize_duration, serialize_duration,
+        sock_addr::SockAddrMarshaler,
+        token_util::{token_fields, TokenParams},
+    },
+};
+
+//const DEFAULT_MAX_TTL: Duration = Duration::from_secs(365*24*60*60 as u64);
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize, Deref, DerefMut)]
+pub struct CertEntry {
+    pub name: String,
+    pub display_name: String,
+    #[serde(serialize_with = "serialize_vec_x509", deserialize_with = "deserialize_vec_x509")]
+    pub certificate: Vec<X509>,
+    pub policies: Vec<String>,
+    #[serde(serialize_with = "serialize_duration", deserialize_with = "deserialize_duration")]
+    pub ttl: Duration,
+    #[serde(serialize_with = "serialize_duration", deserialize_with = "deserialize_duration")]
+    pub max_ttl: Duration,
+    #[serde(serialize_with = "serialize_duration", deserialize_with = "deserialize_duration")]
+    pub period: Duration,
+    pub bound_cidrs: Vec<SockAddrMarshaler>,
+    pub allowed_names: Vec<String>,
+    pub allowed_common_names: Vec<String>,
+    pub allowed_dns_sans: Vec<String>,
+    pub allowed_email_sans: Vec<String>,
+    pub allowed_uri_sans: Vec<String>,
+    pub allowed_organizational_units: Vec<String>,
+    pub allowed_metadata_extensions: Vec<String>,
+    pub required_extensions: Vec<String>,
+    pub ocsp_enabled: bool,
+    #[serde(serialize_with = "serialize_vec_x509", deserialize_with = "deserialize_vec_x509")]
+    pub ocsp_ca_certificates: Vec<X509>,
+    pub ocsp_servers_override: Vec<String>,
+    pub ocsp_fail_open: bool,
+    pub ocsp_query_all_servers: bool,
+    #[serde(flatten)]
+    #[deref]
+    #[deref_mut]
+    pub token_params: TokenParams,
+}
+
+impl CertBackend {
+    pub fn certs_path(&self) -> Path {
+        let cert_backend_ref1 = Arc::clone(&self.inner);
+        let cert_backend_ref2 = Arc::clone(&self.inner);
+        let cert_backend_ref3 = Arc::clone(&self.inner);
```

Comment:
It seems you can just use `Arc::clone(...)` in the places where they need it, thus the 3 local variables can be avoided.

Reply:
> It seems you can just use `Arc::clone(...)` in the places where they need it, thus the 3 local variables can be avoided.

This won't work because these variables are used by calling their member functions rather than using the variables themselves.
![image](https://github.com/user-attachments/assets/3434a619-5c40-4cf6-b020-373abbd44e28)